templates/kubernetes/terraform/modules/kubernetes/files/kratos-values.yml

```diff
@@ -88,20 +95,8 @@ kratos:
           # enabled: true
         password:
           enabled: true
-        oidc: # TODO: allow methods to be configured with parameters
-          enabled: true
-          # config:
-          # TODO: use SELFSERVICE_METHODS_OIDC_CONFIG_PROVIDERS to set via secrets
-          # [{"id":"github","provider":"github","client_id":"id","client_secret":"secret","mapper_url":"http://zombism.ca/github.data-mapper.jsonnet","scope":["user:email"]}]
-            # providers:
-            #   - id: github
-            #     provider: github
-            #     client_id: # TODO: load secret
-            #     client_secret: # TODO : load secret
-            #     # TODO: Change to static or deploy somewhere within infra
-            #     mapper_url: http://zombism.ca/github.data-mapper.jsonnet
-            #     scope:
-            #       - user:email
+        # To set up OIDC (Github, Google login, etc.) this value can be provided via env var, see the deployment section above
+        # oidc:
```

Comment:
does one have to re-enable `oidc: enabled: true` for this to work? or its automatic?  since the env-var is just the list of providers` but not the following part
```yml
oidc: 
  enabled: true
  config: 
     # providers 
```

Reply:
No, this should be left commented out permanently, it was just a placeholder for the comment above it. But yeah, maybe that is counter-intuitive..

Reply:
The env var is the entire `oidc` block

Reply:
>The env var is the entire oidc block

the nesting of properties is different on mine, it goes something like 
```yml
        oidc: 
          enabled: true
          config:
            providers:
              - id: github
                provider: github
                client_id: 
                client_secret: 
                mapper_url: github.data-mapper.jsonnet
                scope:
                  - user:email
```
so the json should be like the following?
```
{"enabled":true,"config":{"providers":[{"id":"github","provider":"github","client_id":"<id>","client_secret":"<secret>","mapper_url":"http://your-url/github.data-mapper.jsonnet","scope":["user:email"]}]}}
```

Reply:
Yes, thanks, I put the wrong one in the docs.

Reply:
looks like the kratos docs’ config is also wrong 😅 
https://www.ory.sh/kratos/docs/concepts/credentials/openid-connect-oidc-oauth2/
has 
```
oidc:
      providers:
        - # REQUIRED - The ID of the provider.
          # DO NOT change this once this is in use.
          id: github
```
but if i do that i get an error 
```
time=2020-11-20T19:19:58Z level=error msg=The provided configuration is invalid and could not be loaded. Check the output below to understand why. audience=application config_file=/etc/config/kratos.yaml service_name=kratos service_version=

selfservice.methods.oidc: map[enabled:false providers:[map[client_id:0983ef056e38a562f29d client_secret:472d3579811c391cc7f992d4eebc45751cf228d3 id:github mapper_url:http://zombism.ca/github.data-mapper.jsonnet provider:github scope:[user:email]]]]
                          ^-- additionalProperties "providers" not allowed
```

Reply:
(reposted because i pasted my clientid in there)
looks like the kratos docs’ config is also wrong 😅 
https://www.ory.sh/kratos/docs/concepts/credentials/openid-connect-oidc-oauth2/
has 
```
oidc:
      providers:
        - # REQUIRED - The ID of the provider.
          # DO NOT change this once this is in use.
          id: github
```
but if i do that i get an error 
```
time=2020-11-20T19:19:58Z level=error msg=The provided configuration is invalid and could not be loaded. Check the output below to understand why. audience=application config_file=/etc/config/kratos.yaml service_name=kratos service_version=

selfservice.methods.oidc: map[enabled:false providers:[map[client_id:client_secret:id:github mapper_url:github.data-mapper.jsonnet provider:github scope:[user:email]]]]
                          ^-- additionalProperties "providers" not allowed
```